XCUITests/ReaderViewUITest.swift

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import XCTest
+
+class ReaderViewTest: BaseTestCase {
+    var navigator: Navigator!
+    var app: XCUIApplication!
+
+    override func setUp() {
+        super.setUp()
+        app = XCUIApplication()
+        navigator = createScreenGraph(app).navigator(self)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+    }
+
+    func testLoadReaderContent() {
+        navigator.goto(BrowserTab)
+        app.buttons["Reader View"].tap()
+        // The settings of reader view are shown as well as the content of the web site
+        waitforExistence(app.buttons["Display Settings"])
+        XCTAssertTrue(app.webViews.staticTexts["The Book of Mozilla"].exists)
+    }
+
+    private func addContentToReaderView() {
+        navigator.goto(BrowserTab)
+        app.buttons["Reader View"].tap()
+        waitforExistence(app.buttons["Add to Reading List"])
+        app.buttons["Add to Reading List"].tap()
+    }
+
+    func testAddToReadingList() {
+        // Initially reading list is empty
+        navigator.goto(NewTabScreen)
+        waitforExistence(app.buttons["HomePanels.ReadingList"])
+        navigator.goto(HomePanel_ReadingList)
+
+        // Check the button is selected (is disabled and the rest bookmarks and so are enabled)
+        XCTAssertFalse(app.buttons["HomePanels.ReadingList"].isEnabled)
+        XCTAssertTrue(app.buttons["HomePanels.Bookmarks"].isEnabled)
+
+        let readingtableTable = app.tables["ReadingTable"]
+        waitforExistence(readingtableTable)
+        let list = app.tables["ReadingTable"].cells.count
+        XCTAssertEqual(list, 0, "There should not be any entry in the reading table")
+
+        navigator.nowAt(NewTabScreen)
+        // Add item to reading list and check that it appears
+        addContentToReaderView()
+        navigator.goto(NewTabScreen)
+        waitforExistence(app.buttons["HomePanels.ReadingList"])
```

Comment:
nvm. That's exactly what is happening. New question. Why do we have to first go to the NewTabscreen cant we go straight to the readinglist?